client/core/core.go

```diff
@@ -4058,8 +4069,18 @@ func (c *Core) connectDEX(acctInfo *db.AccountInfo) (*dexConnection, error) {
 	// If the initial connection returned an error, shut it down to kill the
 	// auto-reconnect cycle.
 	if err != nil {
+		dc := &dexConnection{
+			WsConn:     conn,
+			log:        c.log,
+			connMaster: connMaster,
+			books:      make(map[string]*bookie),
+			acct:       newDEXAccount(acctInfo),
+			trades:     make(map[order.OrderID]*trackedTrade),
+			notify:     c.notify,
+			connected:  0,
+		}
```

Comment:
Rather than repeat this, might as well do this once up above, and just assign to the other fields directly in the non-error case.